crates/uv-resolver/src/resolver/environment.rs

```diff
@@ -449,6 +449,45 @@ impl<'d> ForkingPossibility<'d> {
     }
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct VersionForker;
+
+impl VersionForker {
+    pub(crate) fn fork(
+        requires_python: &VersionSpecifiers,
+        python_requirement: &PythonRequirement,
+        env: &ResolverEnvironment,
+    ) -> Vec<ResolverEnvironment> {
+        let requires_python = RequiresPython::from_specifiers(requires_python);
+        let lower = requires_python.range().lower().clone();
+
+        // Attempt to split the current Python requirement based on the `requires-python` specifier.
+        //
+        // For example, if the current requirement is `>=3.10`, and the split point is `3.11`, then
+        // the result will be `>=3.10 and <3.11` and `>=3.11`.
```

Comment:
Is it worth a `TRACE`-level log message here?

Reply:
Would also be helpful to have a debug log when we hit the `return Ok(Some(ResolverVersion::Forked(forks)));` branch so we log that we fork due to requires python